// @pqrux/simple-pix/lib/models/pix_generator/index.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
const crc16_1 = __importDefault(require("../../utils/crc16"));
const pix_container_1 = __importDefault(require("../pix_container"));
const pix_info_1 = __importDefault(require("../pix_info"));
class PIXGenerator {
    constructor() {
        this.mainContainer = new pix_container_1.default("");
    }
    insertCRC16(text) {
        text = `${text}6304`;
        const code = (0, crc16_1.default)(text);
        return `${text}${code.toUpperCase()}`;
    }
    generateBRCode() {
        let payload = this.mainContainer.toString(true);
        payload = this.insertCRC16(payload);
        return payload;
    }
    setPayloadFormatIndicator() {
        this.mainContainer.addChild(new pix_info_1.default("00", "01", 2));
    }
    setPointOfInitiationMethod(value = "12") {
        this.mainContainer.addChild(new pix_info_1.default("01", value, 2));
    }
    setMerchantAccountInformation(pixKey, GUI = "BR.GOV.BCB.PIX") {
        const mai = new pix_container_1.default("26");
        mai.addChild(new pix_info_1.default("00", GUI));
        mai.addChild(new pix_info_1.default("01", pixKey));
        this.mainContainer.addChild(mai);
    }
    setMerchantCategoryCode(code = "0000") {
        this.mainContainer.addChild(new pix_info_1.default("52", code, 4));
    }
    setTransactionCurrency(code = "986") {
        this.mainContainer.addChild(new pix_info_1.default("53", code, 3));
    }
    setValue(value) {
        this.mainContainer.addChild(new pix_info_1.default("54", value.toFixed(2), 13));
    }
    setCountryCode(code = "BR") {
        this.mainContainer.addChild(new pix_info_1.default("58", code, 2));
    }
    setMerchantName(name) {
        this.mainContainer.addChild(new pix_info_1.default("59", name, 25));
    }
    setMerchantCity(city) {
        this.mainContainer.addChild(new pix_info_1.default("60", city, 15));
    }
    setPostalCode(postal) {
        this.mainContainer.addChild(new pix_info_1.default("61", postal, 99));
    }
    setAdditionalDataFieldTemplate(txID = "***") {
        const adft = new pix_container_1.default("62");
        adft.addChild(new pix_info_1.default("05", txID, 25));
        this.mainContainer.addChild(adft);
    }
}
exports.default = PIXGenerator;
