// pix-utils/dist/main/types/pixDynamicPayload.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.PayloadExample = exports.PixDynamicStatus = void 0;
var PixDynamicStatus;
(function (PixDynamicStatus) {
    PixDynamicStatus["ATIVA"] = "ATIVA";
    PixDynamicStatus["CONCLUIDA"] = "CONCLUIDA";
    PixDynamicStatus["REMOVIDA_PELO_USUARIO_RECEBEDOR"] = "REMOVIDA_PELO_USUARIO_RECEBEDOR";
    PixDynamicStatus["REMOVIDA_PELO_PSP"] = "REMOVIDA_PELO_PSP";
})(PixDynamicStatus = exports.PixDynamicStatus || (exports.PixDynamicStatus = {}));
exports.PayloadExample = {
    txid: 'fc9a4366-ff3d-4964-b5db-c6c91a8722d3',
    revisao: 3,
    calendario: {
        criacao: '2020-09-15T19:39:54.013Z',
        apresentacao: '2020-04-01T18:00:00Z',
        expiracao: 3600,
    },
    status: PixDynamicStatus.ATIVA,
    valor: {
        original: '500.00',
        final: '500.00',
        modalidadeAlteracao: 0,
    },
    chave: '7407c9c8-f78b-11ea-adc1-0242ac120002',
    solicitacaoPagador: 'Informar cartão fidelidade',
    infoAdicionais: [{ nome: 'quantidade', valor: '2' }],
};
