// pix-utils/dist/main/types/pixEmvSchema.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.TagsWithSubTags = exports.EmvMandatory = exports.EmvMaiSchema = exports.EmvAdditionalDataSchema = exports.EmvSchema = void 0;
var EmvSchema;
(function (EmvSchema) {
    EmvSchema[EmvSchema["TAG_INIT"] = 0] = "TAG_INIT";
    EmvSchema[EmvSchema["TAG_ONETIME"] = 1] = "TAG_ONETIME";
    EmvSchema[EmvSchema["TAG_MAI"] = 26] = "TAG_MAI";
    EmvSchema[EmvSchema["TAG_MCC"] = 52] = "TAG_MCC";
    EmvSchema[EmvSchema["TAG_TRANSACTION_CURRENCY"] = 53] = "TAG_TRANSACTION_CURRENCY";
    EmvSchema[EmvSchema["TAG_TRANSACTION_AMOUNT"] = 54] = "TAG_TRANSACTION_AMOUNT";
    EmvSchema[EmvSchema["TAG_COUNTRY_CODE"] = 58] = "TAG_COUNTRY_CODE";
    EmvSchema[EmvSchema["TAG_MERCHANT_NAME"] = 59] = "TAG_MERCHANT_NAME";
    EmvSchema[EmvSchema["TAG_MERCHANT_CITY"] = 60] = "TAG_MERCHANT_CITY";
    EmvSchema[EmvSchema["TAG_ADDITIONAL_DATA"] = 62] = "TAG_ADDITIONAL_DATA";
    EmvSchema[EmvSchema["TAG_CRC"] = 63] = "TAG_CRC";
})(EmvSchema = exports.EmvSchema || (exports.EmvSchema = {}));
var EmvAdditionalDataSchema;
(function (EmvAdditionalDataSchema) {
    EmvAdditionalDataSchema[EmvAdditionalDataSchema["TAG_TXID"] = 5] = "TAG_TXID";
})(EmvAdditionalDataSchema = exports.EmvAdditionalDataSchema || (exports.EmvAdditionalDataSchema = {}));
var EmvMaiSchema;
(function (EmvMaiSchema) {
    EmvMaiSchema[EmvMaiSchema["TAG_MAI_GUI"] = 0] = "TAG_MAI_GUI";
    EmvMaiSchema[EmvMaiSchema["TAG_MAI_PIXKEY"] = 1] = "TAG_MAI_PIXKEY";
    EmvMaiSchema[EmvMaiSchema["TAG_MAI_INFO_ADD"] = 2] = "TAG_MAI_INFO_ADD";
    EmvMaiSchema[EmvMaiSchema["TAG_MAI_FSS"] = 3] = "TAG_MAI_FSS";
    EmvMaiSchema[EmvMaiSchema["TAG_MAI_URL"] = 25] = "TAG_MAI_URL";
    EmvMaiSchema["BC_GUI"] = "br.gov.bcb.pix";
})(EmvMaiSchema = exports.EmvMaiSchema || (exports.EmvMaiSchema = {}));
var EmvMandatory;
(function (EmvMandatory) {
    EmvMandatory[EmvMandatory["TAG_MCC"] = 52] = "TAG_MCC";
    EmvMandatory[EmvMandatory["TAG_TRANSACTION_CURRENCY"] = 53] = "TAG_TRANSACTION_CURRENCY";
    EmvMandatory[EmvMandatory["TAG_COUNTRY_CODE"] = 58] = "TAG_COUNTRY_CODE";
    EmvMandatory[EmvMandatory["TAG_MERCHANT_NAME"] = 59] = "TAG_MERCHANT_NAME";
    EmvMandatory[EmvMandatory["TAG_MERCHANT_CITY"] = 60] = "TAG_MERCHANT_CITY";
})(EmvMandatory = exports.EmvMandatory || (exports.EmvMandatory = {}));
var TagsWithSubTags;
(function (TagsWithSubTags) {
    TagsWithSubTags[TagsWithSubTags["TAG_MAI"] = 26] = "TAG_MAI";
    TagsWithSubTags[TagsWithSubTags["TAG_ADDITIONAL_DATA"] = 62] = "TAG_ADDITIONAL_DATA";
})(TagsWithSubTags = exports.TagsWithSubTags || (exports.TagsWithSubTags = {}));
